projects/js-packages/charts/src/components/line-chart/line-chart.tsx

```diff
@@ -216,341 +235,404 @@ const HighlightTooltip: React.FC< {
+												? `url(#area-gradient-${ chartId }-${ index + 1 })`
+												: 'transparent'
+										}
+										renderLine={ true }
+										curve={ getCurveType( curveType, smoothing ) }
+										lineProps={ lineProps }
+									/>
+								</g>
+							);
+						} ) }
+
+						{ withTooltips && (
+							<>
+								{ dataSorted && (
+									<HighlightTooltip
+										series={ dataSorted }
+										selectedIndex={ selectedIndex !== undefined ? selectedIndex : undefined }
+									/>
+								) }
+								<Tooltip
+									detectBounds
+									snapTooltipToDatumX
+									snapTooltipToDatumY
+									showSeriesGlyphs
+									renderTooltip={ focusableRenderTooltip }
+									renderGlyph={ tooltipRenderGlyph }
 									glyphStyle={ glyphStyle }
+									showVerticalCrosshair={ withTooltipCrosshairs?.showVertical }
+									showHorizontalCrosshair={ withTooltipCrosshairs?.showHorizontal }
 								/>
-							) }
-
-							{ withGradientFill && (
-								<LinearGradient
-									id={ `area-gradient-${ chartId }-${ index + 1 }` }
-									from={ stroke }
-									fromOpacity={ 0.4 }
-									toOpacity={ 0.1 }
-									to={ theme.backgroundColor }
-									{ ...seriesData.options?.gradient }
-									data-testid="line-gradient"
-								/>
-							) }
-							<AreaSeries
-								key={ seriesData?.label }
-								dataKey={ seriesData?.label }
-								data={ seriesData.data as DataPointDate[] }
-								{ ...accessors }
-								fill={
-									withGradientFill
-										? `url(#area-gradient-${ chartId }-${ index + 1 })`
-										: 'transparent'
-								}
-								renderLine={ true }
-								curve={ getCurveType( curveType, smoothing ) }
-								lineProps={ lineProps }
-							/>
-						</g>
-					);
-				} ) }
-
-				{ withTooltips && (
-					<>
-						{ dataSorted && (
-							<HighlightTooltip
-								series={ dataSorted }
-								selectedIndex={ selectedIndex !== undefined ? selectedIndex : undefined }
-							/>
+							</>
 						) }
-						<Tooltip
-							detectBounds
-							snapTooltipToDatumX
-							snapTooltipToDatumY
-							showSeriesGlyphs
-							renderTooltip={ focusableRenderTooltip }
-							renderGlyph={ tooltipRenderGlyph }
-							glyphStyle={ glyphStyle }
-							showVerticalCrosshair={ withTooltipCrosshairs?.showVertical }
-							showHorizontalCrosshair={ withTooltipCrosshairs?.showHorizontal }
+
+						{ /* Component to expose scale data via ref */ }
+						<LineChartInternal
+							chartRef={ internalChartRef }
+							width={ width }
+							height={ height }
+							margin={ margin }
 						/>
-					</>
+					</XYChart>
+				</div>
+
+				{ /* Render annotations as external overlay to avoid interaction blocking */ }
+				{ annotations?.length && (
+					<LineChartAnnotationsOverlay
+						chartRef={ internalChartRef }
+						annotations={ annotations }
+						chartWidth={ width }
+						chartHeight={ height - ( showLegend ? legendHeight : 0 ) }
+					/>
 				) }
```

Comment:
The XYChart renders a `rect` interaction layer over the whole chart which is used for tooltips, etc. If the annotations are rendered within the XYChart then they are not clickable, even with z-index changes. This means we need to render them above and create our own annotations overlay which handles synchronising the chart dimensions and scales for correct positioning of the annotations.